hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/io/TestHoodieTimelineArchiveLog.java

```diff
@@ -183,6 +217,24 @@ public void testArchiveTableWithArchival(boolean enableMetadata) throws Exceptio
     }
   }
 
+  @ParameterizedTest
+  @MethodSource("testArchiveTableWithArchivalCleanUp")
+  public void testArchiveTableWithArchivalCleanUp(boolean enableMetadata, boolean enableArchiveClean, int archiveFilesToKeep) throws Exception {
+    HoodieWriteConfig writeConfig = initTestTableAndGetWriteConfig(enableMetadata, 2, 3, 2, enableArchiveClean, archiveFilesToKeep);
+    for (int i = 1; i < 10; i++) {
+      testTable.doWriteOperation("0000000" + i, WriteOperationType.UPSERT, i == 1 ? Arrays.asList("p1", "p2") : Collections.emptyList(), Arrays.asList("p1", "p2"), 2);
+      // trigger archival
+      archiveAndGetCommitsList(writeConfig);
+    }
+    String archivePath = metaClient.getArchivePath();
+    RemoteIterator<LocatedFileStatus> iter = metaClient.getFs().listFiles(new Path(archivePath), false);
+    ArrayList<LocatedFileStatus> files = new ArrayList<>();
+    while (iter.hasNext()) {
+      files.add(iter.next());
+    }
+    assertEquals(archiveFilesToKeep, files.size());
```

Comment:
Can we also add assertion that earliest files are deleted and not latest ones in archive folder.

Reply:
Changed.